engine/Build.bee.cs

```diff
@@ -624,6 +628,110 @@ static void SetupDependency(NativeProgram np)
             "ICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC"
         });
 
+        np.IncludeDirectories.Add(c => IsWindows(c), new NPath[] {
+             ".",
+            "third_party",
+            "src",
+            flutterRoot,
+            flutterRoot + "/third_party/rapidjson/include",
+            flutterRoot +"/third_party/angle/include",
+            skiaRoot,
+            flutterRoot + "/flutter/third_party/txt/src",
+            flutterRoot + "/third_party/harfbuzz/src",
+            flutterRoot + "/third_party/icu/source/common",
+
+            flutterRoot + "/third_party/icu/source/common",
+            flutterRoot + "/third_party/icu/source/i18n",
+        });
+        np.CompilerSettings().Add(c => IsWindows(c), c => c.WithCustomFlags(new [] {
+            "-D_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING",
+            "-DUSE_OPENSSL=1",
+            "-D__STD_C",
+            "-D_CRT_RAND_S",
+            "-D_CRT_SECURE_NO_DEPRECATE",
+            "-D_HAS_EXCEPTIONS=0",
+            "-D_SCL_SECURE_NO_DEPRECATE",
+            "-DWIN32_LEAN_AND_MEAN",
+            "-DNOMINMAX",
+            "-D_ATL_NO_OPENGL",
+            "-D_WINDOWS",
+            "-DCERT_CHAIN_PARA_HAS_EXTRA_FIELDS",
+            "-DNTDDI_VERSION=0x06030000",
+            "-DPSAPI_VERSION=1",
+            "-DWIN32",
+            "-D_SECURE_ATL",
+            "-D_USING_V110_SDK71_",
+            "-D_UNICODE",
+            "-DUNICODE",
+            "-D_WIN32_WINNT=0x0603",
+            "-DWINVER=0x0603",
+            "-D_DEBUG",
+            "-DU_USING_ICU_NAMESPACE=0",
+            "-DU_ENABLE_DYLOAD=0",
+            "-DUSE_CHROMIUM_ICU=1",
+            "-DU_STATIC_IMPLEMENTATION",
+            "-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_FILE",
+            "-DUCHAR_TYPE=wchar_t",
+            "-DFLUTTER_RUNTIME_MODE_DEBUG=1",
+            "-DFLUTTER_RUNTIME_MODE_PROFILE=2",
+            "-DFLUTTER_RUNTIME_MODE_RELEASE=3",
+            "-DFLUTTER_RUNTIME_MODE_JIT_RELEASE=4",
+            "-DFLUTTER_RUNTIME_MODE=1",
+            "-DFLUTTER_JIT_RUNTIME=1",
+
+            "-DSK_ENABLE_SPIRV_VALIDATION",
+            "-D_CRT_SECURE_NO_WARNINGS",
+            "-D_HAS_EXCEPTIONS=0",
+            "-DWIN32_LEAN_AND_MEAN",
+            "-DNOMINMAX",
+            "-DSK_GAMMA_APPLY_TO_A8",
+            "-DSK_ALLOW_STATIC_GLOBAL_INITIALIZERS=1",
+            // "-DGR_TEST_UTILS=1",
```

Comment:
we should add some comments on why not using this Macro, maybe paste this link (https://github.com/microsoft/vcpkg/issues/12123) as a reference
and add a TODO to find a way to fix this issue by changing some configurations on txt_lib